File: app/src/main/java/techies/it/preschoolmaths/ui/GameLevel1.java
package techies.it.preschoolmaths.ui;

import android.annotation.SuppressLint;
import android.annotation.TargetApi;
import android.content.Context;
import android.graphics.Color;
import android.graphics.Paint;
import android.graphics.PorterDuff;
import android.graphics.PorterDuffXfermode;
import android.graphics.Typeface;
import android.os.Build;
import android.os.Bundle;
import android.util.Log;
import android.view.View;
import android.view.WindowManager;
import android.view.animation.Animation;
import android.view.animation.AnimationUtils;
import android.widget.ImageButton;
import android.widget.ImageView;
import android.widget.RelativeLayout;
import android.widget.TextView;

import androidx.appcompat.app.AppCompatActivity;
import butterknife.Bind;
import butterknife.ButterKnife;
import techies.it.preschoolmaths.R;
import techies.it.preschoolmaths.utils.DrawingView;

public class GameLevel1 extends AppCompatActivity implements View.OnClickListener, Animation.AnimationListener
{

    private static final String TAG = "GameLevel1";

    private int currentApiVersion;
    DrawingView drawingView;
    public Paint mPaint;
    private boolean mBtnEnable = true;

    @Bind(R.id.contentLayoutGameLevel1)
    RelativeLayout mContentLayoutGameLevel1;

    @Bind(R.id.bigBoardLabel)
    TextView bigBoardLabel;

    @Bind(R.id.smallBoardLabel)
    TextView smallBoardLabel;

    @Bind(R.id.numbersTV)
    TextView mNumbers;

    @Bind(R.id.btPlay0)
    ImageButton mPlay0;

    @Bind(R.id.btPlay1)
    ImageButton mPlay1;

    @Bind(R.id.btPlay2)
    ImageButton mPlay2;

    @Bind(R.id.btPlay3)
    ImageButton mPlay3;

    @Bind(R.id.btPlay4)
    ImageButton mPlay4;

    @Bind(R.id.btPlay5)
    ImageButton mPlay5;

    @Bind(R.id.btPlay6)
    ImageButton mPlay6;

    @Bind(R.id.btPlay7)
    ImageButton mPlay7;

    @Bind(R.id.btPlay8)
    ImageButton mPlay8;

    @Bind(R.id.btPlay9)
    ImageButton mPlay9;

    @Bind(R.id.backBT)
    ImageButton mBack;

    @Bind(R.id.star1)
    ImageView mStar1;

    @Bind(R.id.star2)
    ImageView mStar2;

    @Bind(R.id.star3)
    ImageView mStar3;

    @Bind(R.id.star4)
    ImageView mStar4;

    @Bind(R.id.star5)
    ImageView mStar5;

    @Bind(R.id.star6)
    ImageView mStar6;

    @Bind(R.id.star7)
    ImageView mStar7;

    @Bind(R.id.star8)
    ImageView mStar8;

    @Bind(R.id.star9)
    ImageView mStar9;

    @Bind(R.id.cloud0IV)
    ImageView mCloud0;

    @Bind(R.id.cloud6IV)
    ImageView mCloud6;

    @Bind(R.id.cloud7IV)
    ImageView mCloud7;

    @Bind(R.id.cloud1IV)
    ImageView mCloud1;

    @Bind(R.id.cloud3IV)
    ImageView mCloud3;

    @Bind(R.id.sunRaysIV)
    ImageView mSunRays;

    @Bind(R.id.dusterIB)
    ImageButton mDuster;

    @Bind(R.id.chalkIB)
    ImageButton mChalk;

    Animation animation270Sec, animation330Sec, animation390Sec, rotation;

    @TargetApi(Build.VERSION_CODES.KITKAT)
    @Override
    protected void onCreate(Bundle savedInstanceState)
    {
        super.onCreate(savedInstanceState);

        currentApiVersion = android.os.Build.VERSION.SDK_INT;

        final int flags = View.SYSTEM_UI_FLAG_LAYOUT_STABLE
                | View.SYSTEM_UI_FLAG_LAYOUT_HIDE_NAVIGATION
                | View.SYSTEM_UI_FLAG_LAYOUT_FULLSCREEN
                | View.SYSTEM_UI_FLAG_HIDE_NAVIGATION
                | View.SYSTEM_UI_FLAG_FULLSCREEN
                | View.SYSTEM_UI_FLAG_IMMERSIVE_STICKY;

        // This work only for android 4.4+
        if (currentApiVersion >= Build.VERSION_CODES.KITKAT)
        {

            getWindow().getDecorView().setSystemUiVisibility(flags);

            // Code below is to handle presses of Volume up or Volume down.
            // Without this, after pressing volume buttons, the navigation bar will
            // show up and won't hide
            final View decorView = getWindow().getDecorView();
            decorView
                    .setOnSystemUiVisibilityChangeListener(new View.OnSystemUiVisibilityChangeListener()
                    {

                        @Override
                        public void onSystemUiVisibilityChange(int visibility)
                        {
                            if ((visibility & View.SYSTEM_UI_FLAG_FULLSCREEN) == 0)
                            {
                                decorView.setSystemUiVisibility(flags);
                            }
                        }
                    });
        }
        else
        {
            getWindow().setFlags(WindowManager.LayoutParams.FLAG_FULLSCREEN, WindowManager.LayoutParams.FLAG_FULLSCREEN);
        }

        setContentView(R.layout.activity_game_level1);

        ButterKnife.bind(this);



        Typeface typeface = Typeface.createFromAsset(getAssets(), "font/cabin_sketch_bold.ttf");


        smallBoardLabel.setTypeface(typeface);
        mNumbers.setTypeface(typeface);

        mPaint = new Paint();
        mPaint.setAntiAlias(true);
        mPaint.setDither(true);
        mPaint.setColor(Color.WHITE);
        mPaint.setStyle(Paint.Style.STROKE);
        mPaint.setStrokeJoin(Paint.Join.ROUND);
        mPaint.setStrokeCap(Paint.Cap.ROUND);
        brushSize(GameLevel1.this, mPaint);

        drawingView = new DrawingView(this, mPaint);
        drawingView.setBackgroundColor(Color.TRANSPARENT);
        drawingView.setButtonEnableListener(new DrawingView.BtnEnableListener()
        {
            @Override
            public void enableButton(boolean enable)
            {
                mBtnEnable = enable;
            }
        });
        mContentLayoutGameLevel1.addView(drawingView);

        if (getResources().getBoolean(R.bool.isTab))
        {
            // for tab
            bigBoardLabel.setTypeface(typeface);

            animation270Sec = AnimationUtils.loadAnimation(getApplicationContext(),
                    R.anim.translation_left_to_right_270_sec_600);

            animation330Sec = AnimationUtils.loadAnimation(getApplicationContext(),
                    R.anim.translation_left_to_right_330_sec_600);

            animation390Sec = AnimationUtils.loadAnimation(getApplicationContext(),
                    R.anim.translation_left_to_right_390_sec_600);

            rotation = AnimationUtils.loadAnimation(getApplicationContext(),
                    R.anim.rotation_600);

            mCloud0.startAnimation(animation270Sec);
            mCloud1.startAnimation(animation330Sec);
            mCloud3.startAnimation(animation330Sec);
            mCloud6.startAnimation(animation390Sec);
            mCloud7.startAnimation(animation390Sec);
            mSunRays.startAnimation(rotation);
        }
        else
        {
            // for mobile

            animation270Sec = AnimationUtils.loadAnimation(getApplicationContext(),
                    R.anim.translation_left_to_right_270_sec_320);

            animation330Sec = AnimationUtils.loadAnimation(getApplicationContext(),
                    R.anim.translation_left_to_right_330_sec_320);

            animation390Sec = AnimationUtils.loadAnimation(getApplicationContext(),
                    R.anim.translation_left_to_right_390_sec_320);

            rotation = AnimationUtils.loadAnimation(getApplicationContext(),
                    R.anim.rotation_320);

            mCloud0.startAnimation(animation270Sec);
            mCloud1.startAnimation(animation330Sec);
            mCloud3.startAnimation(animation330Sec);
            mCloud6.startAnimation(animation390Sec);
            mCloud7.startAnimation(animation390Sec);
            mSunRays.startAnimation(rotation);
        }

        mPlay0.setOnClickListener(this);
        mPlay1.setOnClickListener(this);
        mPlay2.setOnClickListener(this);
        mPlay3.setOnClickListener(this);
        mPlay4.setOnClickListener(this);
        mPlay5.setOnClickListener(this);
        mPlay6.setOnClickListener(this);
        mPlay7.setOnClickListener(this);
        mPlay8.setOnClickListener(this);
        mPlay9.setOnClickListener(this);
        mDuster.setOnClickListener(this);
        mChalk.setOnClickListener(this);
        mBack.setOnClickListener(this);
    }


    public static void brushSize(Context context, Paint mPaint)
    {
        if (context.getResources().getBoolean(R.bool.isTab))
        {
            mPaint.setStrokeWidth(12);
        }
        else
        {
            mPaint.setStrokeWidth(24);
        }

    }


    @Override
    public void onClick(View v)
    {


        Log.d(TAG, "onClick: ");

        if (mBtnEnable)
        {
            Log.d(TAG, "onClick: enabled true");

            mPaint.setXfermode(null);
            mPaint.setAlpha(0xFF);

            if (v.getId() == R.id.btPlay0)
            {
                mNumbers.setText("0");
                mStar1.setVisibility(View.INVISIBLE);
                mStar2.setVisibility(View.INVISIBLE);
                mStar3.setVisibility(View.INVISIBLE);
                mStar4.setVisibility(View.INVISIBLE);
                mStar5.setVisibility(View.INVISIBLE);
                mStar6.setVisibility(View.INVISIBLE);
                mStar7.setVisibility(View.INVISIBLE);
                mStar8.setVisibility(View.INVISIBLE);
                mStar9.setVisibility(View.INVISIBLE);
                drawingView.startNew();
                mPaint.setColor(Color.WHITE);
                brushSize(GameLevel1.this, mPaint);
            }
            else if (v.getId() == R.id.btPlay1)
            {
                mNumbers.setText("1");
                mStar1.setVisibility(View.INVISIBLE);
                mStar2.setVisibility(View.INVISIBLE);
                mStar3.setVisibility(View.INVISIBLE);
                mStar4.setVisibility(View.INVISIBLE);
                mStar5.setVisibility(View.VISIBLE);
                mStar6.setVisibility(View.INVISIBLE);
                mStar7.setVisibility(View.INVISIBLE);
                mStar8.setVisibility(View.INVISIBLE);
                mStar9.setVisibility(View.INVISIBLE);
                drawingView.startNew();
                mPaint.setColor(Color.WHITE);
                brushSize(GameLevel1.this, mPaint);
            }
            else if (v.getId() == R.id.btPlay2)
            {
                mNumbers.setText("2");
                mStar1.setVisibility(View.INVISIBLE);
                mStar2.setVisibility(View.INVISIBLE);
                mStar3.setVisibility(View.INVISIBLE);
                mStar4.setVisibility(View.VISIBLE);
                mStar5.setVisibility(View.INVISIBLE);
                mStar6.setVisibility(View.VISIBLE);
                mStar7.setVisibility(View.INVISIBLE);
                mStar8.setVisibility(View.INVISIBLE);
                mStar9.setVisibility(View.INVISIBLE);
                drawingView.startNew();
                mPaint.setColor(Color.WHITE);
                brushSize(GameLevel1.this, mPaint);
            }
            else if (v.getId() == R.id.btPlay3)
            {
                mNumbers.setText("3");
                mStar1.setVisibility(View.VISIBLE);
                mStar2.setVisibility(View.INVISIBLE);
                mStar3.setVisibility(View.INVISIBLE);
                mStar4.setVisibility(View.INVISIBLE);
                mStar5.setVisibility(View.VISIBLE);
                mStar6.setVisibility(View.INVISIBLE);
                mStar7.setVisibility(View.INVISIBLE);
                mStar8.setVisibility(View.INVISIBLE);
                mStar9.setVisibility(View.VISIBLE);
                drawingView.startNew();
                mPaint.setColor(Color.WHITE);
                brushSize(GameLevel1.this, mPaint);
            }
            else if (v.getId() == R.id.btPlay4)
            {
                mNumbers.setText("4");
                mStar1.setVisibility(View.INVISIBLE);
                mStar2.setVisibility(View.VISIBLE);
                mStar3.setVisibility(View.INVISIBLE);
                mStar4.setVisibility(View.VISIBLE);
                mStar5.setVisibility(View.INVISIBLE);
                mStar6.setVisibility(View.VISIBLE);
                mStar7.setVisibility(View.INVISIBLE);
                mStar8.setVisibility(View.VISIBLE);
                mStar9.setVisibility(View.INVISIBLE);
                drawingView.startNew();
                mPaint.setColor(Color.WHITE);
                brushSize(GameLevel1.this, mPaint);
            }
            else if (v.getId() == R.id.btPlay5)
            {
                mNumbers.setText("5");
                mStar1.setVisibility(View.INVISIBLE);
                mStar2.setVisibility(View.VISIBLE);
                mStar3.setVisibility(View.INVISIBLE);
                mStar4.setVisibility(View.VISIBLE);
                mStar5.setVisibility(View.VISIBLE);
                mStar6.setVisibility(View.VISIBLE);
                mStar7.setVisibility(View.INVISIBLE);
                mStar8.setVisibility(View.VISIBLE);
                mStar9.setVisibility(View.INVISIBLE);
                drawingView.startNew();
                mPaint.setColor(Color.WHITE);
                brushSize(GameLevel1.this, mPaint);
            }
            else if (v.getId() == R.id.btPlay6)
            {
                mNumbers.setText("6");
                mStar1.setVisibility(View.VISIBLE);
                mStar2.setVisibility(View.INVISIBLE);
                mStar3.setVisibility(View.VISIBLE);
                mStar4.setVisibility(View.VISIBLE);
                mStar5.setVisibility(View.INVISIBLE);
                mStar6.setVisibility(View.VISIBLE);
                mStar7.setVisibility(View.VISIBLE);
                mStar8.setVisibility(View.INVISIBLE);
                mStar9.setVisibility(View.VISIBLE);
                drawingView.startNew();
                mPaint.setColor(Color.WHITE);
                brushSize(GameLevel1.this, mPaint);
            }
            else if (v.getId() == R.id.btPlay7)
            {
                mNumbers.setText("7");
                mStar1.setVisibility(View.VISIBLE);
                mStar2.setVisibility(View.VISIBLE);
                mStar3.setVisibility(View.VISIBLE);
                mStar4.setVisibility(View.INVISIBLE);
                mStar5.setVisibility(View.VISIBLE);
                mStar6.setVisibility(View.INVISIBLE);
                mStar7.setVisibility(View.VISIBLE);
                mStar8.setVisibility(View.VISIBLE);
                mStar9.setVisibility(View.VISIBLE);
                drawingView.startNew();
                mPaint.setColor(Color.WHITE);
                brushSize(GameLevel1.this, mPaint);
            }
            else if (v.getId() == R.id.btPlay8)
            {
                mNumbers.setText("8");
                mStar1.setVisibility(View.VISIBLE);
                mStar2.setVisibility(View.VISIBLE);
                mStar3.setVisibility(View.VISIBLE);
                mStar4.setVisibility(View.VISIBLE);
                mStar5.setVisibility(View.INVISIBLE);
                mStar6.setVisibility(View.VISIBLE);
                mStar7.setVisibility(View.VISIBLE);
                mStar8.setVisibility(View.VISIBLE);
                mStar9.setVisibility(View.VISIBLE);
                drawingView.startNew();
                mPaint.setColor(Color.WHITE);
                brushSize(GameLevel1.this, mPaint);
            }
            else if (v.getId() == R.id.btPlay9)
            {
                mNumbers.setText("9");
                mStar1.setVisibility(View.VISIBLE);
                mStar2.setVisibility(View.VISIBLE);
                mStar3.setVisibility(View.VISIBLE);
                mStar4.setVisibility(View.VISIBLE);
                mStar5.setVisibility(View.VISIBLE);
                mStar6.setVisibility(View.VISIBLE);
                mStar7.setVisibility(View.VISIBLE);
                mStar8.setVisibility(View.VISIBLE);
                mStar9.setVisibility(View.VISIBLE);
                drawingView.startNew();
                mPaint.setColor(Color.WHITE);
                brushSize(GameLevel1.this, mPaint);
            }
            else if (v.getId() == R.id.dusterIB)
            {
                mPaint.setStrokeWidth(40);
                mPaint.setXfermode(new PorterDuffXfermode(PorterDuff.Mode.CLEAR));
            }
            else if (v.getId() == R.id.chalkIB)
            {
                mPaint.setColor(Color.WHITE);
                brushSize(GameLevel1.this, mPaint);
            }
            else if (v.getId() == R.id.backBT)
            {
                finish();
            }
        }

    }

    @SuppressLint("NewApi")
    @Override
    public void onWindowFocusChanged(boolean hasFocus)
    {
        super.onWindowFocusChanged(hasFocus);
        if (currentApiVersion >= Build.VERSION_CODES.KITKAT && hasFocus)
        {
            getWindow().getDecorView().setSystemUiVisibility(
                    View.SYSTEM_UI_FLAG_LAYOUT_STABLE
                            | View.SYSTEM_UI_FLAG_LAYOUT_HIDE_NAVIGATION
                            | View.SYSTEM_UI_FLAG_LAYOUT_FULLSCREEN
                            | View.SYSTEM_UI_FLAG_HIDE_NAVIGATION
                            | View.SYSTEM_UI_FLAG_FULLSCREEN
                            | View.SYSTEM_UI_FLAG_IMMERSIVE_STICKY);
        }
    }

    @Override
    public void onAnimationStart(Animation animation)
    {

    }

    @Override
    public void onAnimationEnd(Animation animation)
    {

    }

    @Override
    public void onAnimationRepeat(Animation animation)
    {

    }
}

File: app/src/main/java/techies/it/preschoolmaths/ui/GameLevel1.kt
package techies.it.preschoolmaths.ui

import androidx.appcompat.app.AppCompatActivity
import android.view.animation.Animation
import techies.it.preschoolmaths.utils.DrawingView
import techies.it.preschoolmaths.R
import android.os.Build
import android.os.Bundle
import butterknife.ButterKnife
import android.annotation.SuppressLint
import android.content.Context
import android.graphics.*
import android.util.Log
import android.view.View
import android.view.animation.AnimationUtils
import androidx.core.view.WindowCompat
import androidx.core.view.WindowInsetsCompat
import androidx.core.view.WindowInsetsControllerCompat
import techies.it.preschoolmaths.databinding.ActivityGameLevel1Binding

class GameLevel1 : AppCompatActivity(), View.OnClickListener, Animation.AnimationListener {
    private lateinit var binding: ActivityGameLevel1Binding
    var view: View?=null
    var drawingView: DrawingView? = null
    var mPaint: Paint? = null
    private var mBtnEnable = true
    var animation270Sec: Animation? = null
    var animation330Sec: Animation? = null
    var animation390Sec: Animation? = null
    var rotation: Animation? = null

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        binding = ActivityGameLevel1Binding.inflate(layoutInflater)
        view = binding.root
        setContentView(view)
        
        mPaint = Paint()
        mPaint?.isAntiAlias = true
        mPaint?.isDither = true
        mPaint?.color = Color.WHITE
        mPaint?.style = Paint.Style.STROKE
        mPaint?.strokeJoin = Paint.Join.ROUND
        mPaint?.strokeCap = Paint.Cap.ROUND
        brushSize(this@GameLevel1, mPaint)
        drawingView = DrawingView(this, mPaint)
        drawingView?.setBackgroundColor(Color.TRANSPARENT)
        drawingView?.setButtonEnableListener { enable -> mBtnEnable = enable }
        
        binding.contentLayoutGameLevel1.addView(drawingView)
        if (resources.getBoolean(R.bool.isTab)) {
            // for tab
            animation270Sec = AnimationUtils.loadAnimation(
                applicationContext,
                R.anim.translation_left_to_right_270_sec_600
            )
            animation330Sec = AnimationUtils.loadAnimation(
                applicationContext,
                R.anim.translation_left_to_right_330_sec_600
            )
            animation390Sec = AnimationUtils.loadAnimation(
                applicationContext,
                R.anim.translation_left_to_right_390_sec_600
            )
            rotation = AnimationUtils.loadAnimation(
                applicationContext,
                R.anim.rotation_600
            )
            binding.cloud0IV.startAnimation(animation270Sec)
            binding.cloud1IV.startAnimation(animation330Sec)
            binding.cloud3IV.startAnimation(animation330Sec)
            binding.cloud6IV.startAnimation(animation390Sec)
            binding.cloud7IV.startAnimation(animation390Sec)
            binding.sunRaysIV.startAnimation(rotation)
        } else {
            // for mobile
            /*animation270Sec = AnimationUtils.loadAnimation(
                applicationContext,
                R.anim.translation_left_to_right_270_sec_320
            )
            animation330Sec = AnimationUtils.loadAnimation(
                applicationContext,
                R.anim.translation_left_to_right_330_sec_320
            )
            animation390Sec = AnimationUtils.loadAnimation(
                applicationContext,
                R.anim.translation_left_to_right_390_sec_320
            )*/
            rotation = AnimationUtils.loadAnimation(
                applicationContext,
                R.anim.rotation_320
            )
            binding.sunRaysIV.startAnimation(rotation)
        }
        binding.btPlay0.setOnClickListener(this)
        binding.btPlay1.setOnClickListener(this)
        binding.btPlay2.setOnClickListener(this)
        binding.btPlay3.setOnClickListener(this)
        binding.btPlay4.setOnClickListener(this)
        binding.btPlay5.setOnClickListener(this)
        binding.btPlay6.setOnClickListener(this)
        binding.btPlay7.setOnClickListener(this)
        binding.btPlay8.setOnClickListener(this)
        binding.btPlay9.setOnClickListener(this)
        binding.dusterIB.setOnClickListener(this)
        binding.chalkIB.setOnClickListener(this)
        binding.backBT.setOnClickListener(this)
    }

    override fun onClick(v: View) {
        Log.d(TAG, "onClick: ")
        if (mBtnEnable) {
            Log.d(TAG, "onClick: enabled true")
            mPaint?.xfermode = null
            mPaint?.alpha = 0xFF
            if (v.id == R.id.btPlay0) {
                binding.numbersTV.text = "0"
                binding.star1.visibility = View.INVISIBLE
                binding.star2.visibility = View.INVISIBLE
                binding.star3.visibility = View.INVISIBLE
                binding.star4.visibility = View.INVISIBLE
                binding.star5.visibility = View.INVISIBLE
                binding.star6.visibility = View.INVISIBLE
                binding.star7.visibility = View.INVISIBLE
                binding.star8.visibility = View.INVISIBLE
                binding.star9.visibility = View.INVISIBLE
                drawingView?.startNew()
                mPaint?.color = Color.WHITE
                brushSize(this@GameLevel1, mPaint)
            } else if (v.id == R.id.btPlay1) {
                binding.numbersTV.text = "1"
                binding.star1.visibility = View.INVISIBLE
                binding.star2.visibility = View.INVISIBLE
                binding.star3.visibility = View.INVISIBLE
                binding.star4.visibility = View.INVISIBLE
                binding.star5.visibility = View.VISIBLE
                binding.star6.visibility = View.INVISIBLE
                binding.star7.visibility = View.INVISIBLE
                binding.star8.visibility = View.INVISIBLE
                binding.star9.visibility = View.INVISIBLE
                drawingView?.startNew()
                mPaint?.color = Color.WHITE
                brushSize(this@GameLevel1, mPaint)
            } else if (v.id == R.id.btPlay2) {
                binding.numbersTV.text = "2"
                binding.star1.visibility = View.INVISIBLE
                binding.star2.visibility = View.INVISIBLE
                binding.star3.visibility = View.INVISIBLE
                binding.star4.visibility = View.VISIBLE
                binding.star5.visibility = View.INVISIBLE
                binding.star6.visibility = View.VISIBLE
                binding.star7.visibility = View.INVISIBLE
                binding.star8.visibility = View.INVISIBLE
                binding.star9.visibility = View.INVISIBLE
                drawingView?.startNew()
                mPaint?.color = Color.WHITE
                brushSize(this@GameLevel1, mPaint)
            } else if (v.id == R.id.btPlay3) {
                binding.numbersTV.text = "3"
                binding.star1.visibility = View.VISIBLE
                binding.star2.visibility = View.INVISIBLE
                binding.star3.visibility = View.INVISIBLE
                binding.star4.visibility = View.INVISIBLE
                binding.star5.visibility = View.VISIBLE
                binding.star6.visibility = View.INVISIBLE
                binding.star7.visibility = View.INVISIBLE
                binding.star8.visibility = View.INVISIBLE
                binding.star9.visibility = View.VISIBLE
                drawingView?.startNew()
                mPaint?.color = Color.WHITE
                brushSize(this@GameLevel1, mPaint)
            } else if (v.id == R.id.btPlay4) {
                binding.numbersTV.text = "4"
                binding.star1.visibility = View.INVISIBLE
                binding.star2.visibility = View.VISIBLE
                binding.star3.visibility = View.INVISIBLE
                binding.star4.visibility = View.VISIBLE
                binding.star5.visibility = View.INVISIBLE
                binding.star6.visibility = View.VISIBLE
                binding.star7.visibility = View.INVISIBLE
                binding.star8.visibility = View.VISIBLE
                binding.star9.visibility = View.INVISIBLE
                drawingView?.startNew()
                mPaint?.color = Color.WHITE
                brushSize(this@GameLevel1, mPaint)
            } else if (v.id == R.id.btPlay5) {
                binding.numbersTV.text = "5"
                binding.star1.visibility = View.INVISIBLE
                binding.star2.visibility = View.VISIBLE
                binding.star3.visibility = View.INVISIBLE
                binding.star4.visibility = View.VISIBLE
                binding.star5.visibility = View.VISIBLE
                binding.star6.visibility = View.VISIBLE
                binding.star7.visibility = View.INVISIBLE
                binding.star8.visibility = View.VISIBLE
                binding.star9.visibility = View.INVISIBLE
                drawingView?.startNew()
                mPaint?.color = Color.WHITE
                brushSize(this@GameLevel1, mPaint)
            } else if (v.id == R.id.btPlay6) {
                binding.numbersTV.text = "6"
                binding.star1.visibility = View.VISIBLE
                binding.star2.visibility = View.INVISIBLE
                binding.star3.visibility = View.VISIBLE
                binding.star4.visibility = View.VISIBLE
                binding.star5.visibility = View.INVISIBLE
                binding.star6.visibility = View.VISIBLE
                binding.star7.visibility = View.VISIBLE
                binding.star8.visibility = View.INVISIBLE
                binding.star9.visibility = View.VISIBLE
                drawingView?.startNew()
                mPaint?.color = Color.WHITE
                brushSize(this@GameLevel1, mPaint)
            } else if (v.id == R.id.btPlay7) {
                binding.numbersTV.text = "7"
                binding.star1.visibility = View.VISIBLE
                binding.star2.visibility = View.VISIBLE
                binding.star3.visibility = View.VISIBLE
                binding.star4.visibility = View.INVISIBLE
                binding.star5.visibility = View.VISIBLE
                binding.star6.visibility = View.INVISIBLE
                binding.star7.visibility = View.VISIBLE
                binding.star8.visibility = View.VISIBLE
                binding.star9.visibility = View.VISIBLE
                drawingView?.startNew()
                mPaint?.color = Color.WHITE
                brushSize(this@GameLevel1, mPaint)
            } else if (v.id == R.id.btPlay8) {
                binding.numbersTV.text = "8"
                binding.star1.visibility = View.VISIBLE
                binding.star2.visibility = View.VISIBLE
                binding.star3.visibility = View.VISIBLE
                binding.star4.visibility = View.VISIBLE
                binding.star5.visibility = View.INVISIBLE
                binding.star6.visibility = View.VISIBLE
                binding.star7.visibility = View.VISIBLE
                binding.star8.visibility = View.VISIBLE
                binding.star9.visibility = View.VISIBLE
                drawingView?.startNew()
                mPaint?.color = Color.WHITE
                brushSize(this@GameLevel1, mPaint)
            } else if (v.id == R.id.btPlay9) {
                binding.numbersTV.text = "9"
                binding.star1.visibility = View.VISIBLE
                binding.star2.visibility = View.VISIBLE
                binding.star3.visibility = View.VISIBLE
                binding.star4.visibility = View.VISIBLE
                binding.star5.visibility = View.VISIBLE
                binding.star6.visibility = View.VISIBLE
                binding.star7.visibility = View.VISIBLE
                binding.star8.visibility = View.VISIBLE
                binding.star9.visibility = View.VISIBLE
                drawingView?.startNew()
                mPaint?.color = Color.WHITE
                brushSize(this@GameLevel1, mPaint)
            } else if (v.id == R.id.dusterIB) {
                mPaint?.strokeWidth = 40f
                mPaint?.xfermode = PorterDuffXfermode(PorterDuff.Mode.CLEAR)
            } else if (v.id == R.id.chalkIB) {
                mPaint?.color = Color.WHITE
                brushSize(this@GameLevel1, mPaint)
            } else if (v.id == R.id.backBT) {
                finish()
            }
        }
    }

    @SuppressLint("NewApi")
    override fun onWindowFocusChanged(hasFocus: Boolean) {
        super.onWindowFocusChanged(hasFocus)
        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.R) {
            window.decorView.windowInsetsController?.hide(
                android.view.WindowInsets.Type.statusBars()
                        or android.view.WindowInsets.Type.navigationBars()
            )
        }else{
            WindowCompat.setDecorFitsSystemWindows(window, false)
            view?.let {
                WindowInsetsControllerCompat(window, it).let { controller ->
                    controller.hide(WindowInsetsCompat.Type.systemBars())
                    controller.systemBarsBehavior = WindowInsetsControllerCompat.BEHAVIOR_SHOW_TRANSIENT_BARS_BY_SWIPE
                }
            }
        }
    }

    override fun onAnimationStart(animation: Animation) {}
    override fun onAnimationEnd(animation: Animation) {}
    override fun onAnimationRepeat(animation: Animation) {}

    companion object {
        private const val TAG = "GameLevel1"
        @JvmStatic
        fun brushSize(context: Context, mPaint: Paint?) {
            if (context.resources.getBoolean(R.bool.isTab)) {
                mPaint?.strokeWidth = 12f
            } else {
                mPaint?.strokeWidth = 24f
            }
        }
    }
}